src/main/java/com/synopsys/integration/alert/provider/blackduck/issues/BlackDuckProviderIssueHandler.java

```diff
@@ -0,0 +1,89 @@
+            issueRequestModel.setIssueCreatedAt(currentDate);
+            issueRequestModel.setIssueUpdatedAt(null);
+            postIssue(bomComponentVersionIssuesUrl, issueRequestModel);
+        }
+    }
+
+    // TODO fix this logic once the bomComponentVersionIssuesUrl supports GET requests directly
+    private Optional<IssueView> retrieveExistingIssue(String bomComponentVersionIssuesUrl, String issueKey) throws IntegrationException {
+        String issueLookupUrl = createIssueLookupUrl(bomComponentVersionIssuesUrl);
+        Request.Builder requestBuilder = RequestFactory.createCommonGetRequestBuilder(issueLookupUrl)
+                                             .addAdditionalHeader("Accept", ISSUE_ENDPOINT_MEDIA_TYPE_V6);
+
+        // This is really a List<BomComponentIssueView>, but BomComponentIssueView is not considered a BlackDuckResponse.
+        List<IssueView> bomComponentIssues = blackDuckService.getAllResponses(requestBuilder, IssueView.class);
+        return bomComponentIssues
+                   .stream()
+                   .filter(issue -> issue.getIssueId().equals(issueKey))
+                   .findAny();
+    }
+
+    private void postIssue(String uri, IssueView requestModel) throws IntegrationException {
+        blackDuckService.post(uri, requestModel);
+    }
+
+    private void putIssue(String uri, IssueView requestModel) throws IntegrationException {
+        ResourceMetadata resourceMetadata = new ResourceMetadata();
+        resourceMetadata.setHref(uri);
+        requestModel.setMeta(resourceMetadata);
+
+        blackDuckService.put(requestModel);
+    }
+
+    private IssueView createIssueRequestModel(BlackDuckProviderIssueModel issueModel) {
+        IssueView blackDuckIssueView = new IssueView();
+        blackDuckIssueView.setIssueId(issueModel.getKey());
+        blackDuckIssueView.setIssueLink(issueModel.getLink());
+        blackDuckIssueView.setIssueAssignee(issueModel.getAssignee());
+        blackDuckIssueView.setIssueStatus(issueModel.getStatus());
+        blackDuckIssueView.setIssueDescription(issueModel.getSummary());
+        return blackDuckIssueView;
+    }
+
+    /**
+     * @param bomComponentVersionIssuesUrl a string in the format of<br/>
+     *                                     {blackDuckUrl}/api/projects/{projectId}/versions/{projectVersionId}/components/{componentId}/issues<br/>
+     *                                     or<br/>
+     *                                     {blackDuckUrl}/api/projects/{projectId}/versions/{projectVersionId}/components/{componentId}/component-versions/{componentVersionId}/issues
+     * @return a string in the format of {blackDuckUrl}/api/projects/{projectId}/versions/{projectVersionId}/issues
+     */
+    private String createIssueLookupUrl(String bomComponentVersionIssuesUrl) {
+        String projectVersionUrl = StringUtils.substringBefore(bomComponentVersionIssuesUrl, "/components");
+        return projectVersionUrl + "/issues";
+    }
+
+}
```

Comment:
This is slightly concerning to me as we're making assumptions here and assume these URLs won't change. Is there no concrete way for us to get this issues url from Blackduck?

Reply:
Not without asking a lot more of Events. This will go away when Black Duck supports GETs against the `bomComponentVersionIssuesUrl`.